packages/core/data-transfer/lib/providers/local-file-destination-provider/__tests__/index.test.ts

```diff
@@ -0,0 +1,174 @@
+      const providerOptions = {
+        encryption: { enabled: false },
+        compression: { enabled: true },
+        file: { path: filePath },
+      };
+      const provider = createLocalFileDestinationProvider(providerOptions);
+
+      await provider.bootstrap();
+
+      expect(provider.results.file.path).toEqual(`${filePath}.tar.gz`);
+    });
+
+    it('Adds .gpg extension to the archive path when encryption is enabled', async () => {
+      const providerOptions: ILocalFileDestinationProviderOptions = {
+        encryption: { enabled: true, key: 'key' },
+        compression: { enabled: false },
+        file: { path: filePath },
+      };
+      const provider = createLocalFileDestinationProvider(providerOptions);
+
+      await provider.bootstrap();
+
+      expect(provider.results.file.path).toEqual(`${filePath}.tar.gpg`);
+    });
+
+    it('Adds .gz.gpg extension to the archive path when encryption and compression are enabled', async () => {
+      const providerOptions: ILocalFileDestinationProviderOptions = {
+        encryption: { enabled: true, key: 'key' },
+        compression: { enabled: true },
+        file: { path: filePath },
+      };
+      const provider = createLocalFileDestinationProvider(providerOptions);
+
+      await provider.bootstrap();
+
+      expect(provider.results.file.path).toEqual(`${filePath}.tar.gz.gpg`);
+    });
+
+    it('Adds the compression step to the stream chain when compression is enabled', async () => {
+      const providerOptions: ILocalFileDestinationProviderOptions = {
+        encryption: { enabled: true, key: 'key' },
+        compression: { enabled: true },
+        file: { path: filePath },
+      };
+      const provider = createLocalFileDestinationProvider(providerOptions);
+      jest.spyOn(provider, 'createGzip');
+
+      await provider.bootstrap();
+
+      expect(provider.createGzip).toHaveBeenCalled();
+    });
+
+    it('Adds the encryption step to the stream chain when encryption is enabled', async () => {
+      jest.spyOn(encryption, 'createEncryptionCipher');
+      const key = 'key';
+      const providerOptions: ILocalFileDestinationProviderOptions = {
+        encryption: { enabled: true, key },
+        compression: { enabled: true },
+        file: { path: filePath },
+      };
+      const provider = createLocalFileDestinationProvider(providerOptions);
+
+      await provider.bootstrap();
+
+      expect(encryption.createEncryptionCipher).toHaveBeenCalledWith(key);
+    });
+  });
+
+  describe('Streaming entities', () => {
+    it('Creates a tar entry stream', () => {
+      const providerOptions: ILocalFileDestinationProviderOptions = {
+        encryption: { enabled: false },
+        compression: { enabled: false },
+        file: { path: filePath },
+      };
+
+      const provider = createLocalFileDestinationProvider(providerOptions);
+      (createTarEntryStream as jest.Mock).mockImplementation(jest.fn());
+      provider.getEntitiesStream();
```

Comment:
Can we check that it actually returns a stream?

Reply:
(same for all of these)

Reply:
Done :)